artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -1382,20 +1382,26 @@ public final Object getObjectProperty(String key) {
             return AMQPMessageIdHelper.INSTANCE.toCorrelationIdString(properties.getCorrelationId());
          }
       } else {
-         Object value = getApplicationPropertiesMap(false).get(key);
-         if (value instanceof UnsignedInteger ||
-             value instanceof UnsignedByte ||
-             value instanceof UnsignedLong ||
-             value instanceof UnsignedShort) {
-            return ((Number)value).longValue();
-         } else {
-            return value;
-         }
+         return getApplicationObjectProperty(key);
       }
 
       return null;
    }
 
+   private Object getApplicationObjectProperty(String key) {
+      Object value = getApplicationPropertiesMap(false).get(key);
+      if (value instanceof Number) {
```

Comment:
Can't we just do the ultimate simplification? I know this would be a semantic change but looking at the usages I only see this being used on filter.

I think it would be safe to change this to:

```java
      if (value instanceof Number) {
             return ((Number) value).longValue();
       }
       return value;
```

Reply:
What if number is float or double

Reply:
hmmm.. yeah.. you're right.. ok!